lizard_connector/connector.py

```diff
@@ -0,0 +1,342 @@
+# coding=utf-8
+"""
+Connector to Lizard api (e.g. https://demo.lizard.net/api/v2) for python.
+
+Includes:
+- Endpoints (Lizard api endoints, e.g. TimeseriesEndpoint)
+- Fetcher (http handling)
+- queryfunctions for special cases such as geographical queries and time
+related queries other queries can be input as a dictionary
+- parserfunctions to parse the json obtained from Endpoint queries
+"""
+
+import json
+import urllib.request
+import urllib.parse
+
+from lizard_connector.query import QueryDictionary
+
+
+class LizardApiTooManyResults(Exception):
+    pass
+
+
+class Fetcher(object):
+
+    def __init__(self, endpoint, base="https://demo.lizard.net",
+                 max_results=1000, username=None, password=None,
+                 all_pages=True):
+        """
+        :param max_results: maximum number of results allowed from one get.
+        :param username: lizard-api user name to log in. Without one no login
+            is used.
+        :param password: lizard-api password to log in. Without one no login
+            is used.
+        :param all_pages: when set to True, on get all pages are obtained.
+            When set to False only the first page is obtained on get.
+        """
+        self.all_pages = all_pages
+        self.max_results = max_results
+        self.next_url = None
+        self.count = None
+        self.username = username
+        self.password = password
+
+    def get(self, url):
+        """
+        GET a json from the api.
+        :param url: Lizard-api valid url.
+        :return: a dictionary of the 'results'-part of the api-response.
+        """
+        json_ = self.request(url)
+        self.count = json_.get('count')
+        self.next_url = json_.get('next')
```

Comment:
Zoals eerder uitgelegd. Het is de keuze om alle afhandeling van 1 query hier te doen. die next_url heb ik nodig voor de iterator. Ik ga de manier van OOP nog met Berto en/of Arjen bespreken (wat willen zijn vanuit Lizard).